hazelcast-client-new/src/main/java/com/hazelcast/client/impl/ClusterAuthenticator.java

```diff
@@ -80,6 +81,15 @@ public void authenticate(ClientConnection connection) throws AuthenticationExcep
         }
         ClientAuthenticationCodec.ResponseParameters result = ClientAuthenticationCodec.decodeResponse(response);
 
-        connection.setRemoteEndpoint(result.address);
+        AuthenticationStatus authenticationStatus = AuthenticationStatus.findByValue(result.status);
+        switch (authenticationStatus) {
+            case AUTHENTICATED:
+                connection.setRemoteEndpoint(result.address);
+                return;
+            case CREDENTIALS_FAILED:
+                throw new AuthenticationException("Invalid credentials!");
+            default:
+                throw new AuthenticationException("Authentication status code not supported.");
+        }
```

Comment:
ClusterAuthenticator does not need that because it is only used to login to an already authenticated clusters.